File: Problem/BOJ_9342_염색체/main.js
const fs = require('fs')
const input = fs.readFileSync('./input.txt').toString().split("\n").map(elem => elem.trim())
const [numOfChromosomes, ...chromosomes] = input

function checkPatterns (chromosome) {
    const pattern = /^[A-F]?A+F+C+[A-F]?$/
    const result = chromosome.match(pattern)

    if (result) {
        console.log('Infected!')
    } else {
        console.log('Good')
    }
}

chromosomes.forEach(chromosome => {
    checkPatterns(chromosome)
})

File: Problem/BOJ_9342_염색체/main.py
from re import compile


def main():
    stdin = open('./input.txt', 'r')
    num_of_chromosomes = int(stdin.readline())

    pattern = compile('^[A-F]?A+F+C+[A-F]?$')

    for _ in range(num_of_chromosomes):
        chromosome = stdin.readline().strip()

        result = pattern.match(chromosome)

        if result is None:
            print('Good')
        else:
            print('Infected!')


if __name__ == '__main__':
    main()